android/src/main/java/com/oney/WebRTCModule/videoEffect/ProcessorProvider.java

```diff
@@ -0,0 +1,26 @@
+package com.oney.WebRTCModule.videoEffect;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ProcessorProvider {
+    private static Map<String, VideoFrameProcessorFactoryInterface> methodMap = new HashMap<String, VideoFrameProcessorFactoryInterface>();
+
+    public static VideoFrameProcessor getProcessor(String name) {
+        if (methodMap.containsKey(name)) {
+            return methodMap.get(name).build();
+        } else
+            return null;
+    }
+
+    public static void addProcessor(String name,
+            VideoFrameProcessorFactoryInterface videoFrameProcessorFactoryInterface) {
+        if (name != null && videoFrameProcessorFactoryInterface != null) {
```

Comment:
Check if there is already an effect registered and throw an exception. Also throw if either are null.

Reply:
```
public static void addProcessor(String name,
            VideoFrameProcessorFactoryInterface videoFrameProcessorFactoryInterface) {
        if (name != null && videoFrameProcessorFactoryInterface != null) {
            if(methodMap.containsKey(name)){
                throw new RuntimeException("VideoFrameProcessorFactry with name "+name" already exists");
            }
            methodMap.put(name, videoFrameProcessorFactoryInterface);
        } else
            throw new NullPointerException("Name or VideoFrameProcessorFactry can not be null");

    }

```

is this look good  for 'already an effect registered and throw an exception' issue